### ISSUE TITLE
Windows R

### ISSUE THREAD

**Issue by kelelexu** [dmlc/mxnet#714]
When I try to use my own data for the test. 
I have met the problem as follows.

'mx.ctx.internal.default.value' not found
OS: Windows 8
Language: R

I am new to this this field, any help is welcome.

**Comment by thirdwing** [comment]
@kelelexu Can you provide a reproducible case for this?

**Comment by kelelexu** [comment]
[code.txt](https://github.com/dmlc/mxnet/files/44850/code.txt)


Auto detect layout of input matrix, use rowmajor..
 Show Traceback
 
 Rerun with Debug
 Error in mx.ctx.default() : 
  object 'mx.ctx.internal.default.value' not found

**Comment by thirdwing** [comment]
Can you try to not use mxnet training in a for loop? It should work.

Error in a for loop or in a function is a known issue on some version of R.

**Comment by kelelexu** [comment]
OK, I will have a try and give a feedback later. Thanks so much.


2015-11-26 16:39 GMT+01:00 Qiang Kou (KK) <notifications@github.com>:

> Can you try to not use mxnet training in a for loop? It should work.
>
> Error in a for loop or in a function is a known issue on some version of R.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dmlc/mxnet/issues/714#issuecomment-159943291>.
>



-- 
Kele Xu, PhD Student of Université Pierre et Marie CURIE
Institute Langevin(Vauquelin Campus)
10, Rue Vauquelin
75231 PARIS Cedex 05
FRANCE

**Comment by kelelexu** [comment]
Sorry to disturb again. I have delete the loop for training, but the issue
remains.

model <- mx.mlp(data.matrix(x[nonval.ind,]), y[nonval.ind],
hidden_node=c(400,100),
+                   out_node=2, out_activation="softmax",
+                   num.round=12, array.batch.size=50,
+                   learning.rate=0.01, momentum=0.5,
+                   dropout = 0.5, activation = "tanh",
+                   eval.data = val.dat, eval.metric=mx.metric.accuracy)
Auto detect layout of input matrix, use rowmajor..
Error in mx.ctx.default() :
  object 'mx.ctx.internal.default.value' not found

2015-11-26 16:46 GMT+01:00 Kele XU <kelele.xu@gmail.com>:

> OK, I will have a try and give a feedback later. Thanks so much.
>
>
> 2015-11-26 16:39 GMT+01:00 Qiang Kou (KK) <notifications@github.com>:
>
>> Can you try to not use mxnet training in a for loop? It should work.
>>
>> Error in a for loop or in a function is a known issue on some version of
>> R.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/dmlc/mxnet/issues/714#issuecomment-159943291>.
>>
>
>
>
> --
> Kele Xu, PhD Student of Université Pierre et Marie CURIE
> Institute Langevin(Vauquelin Campus)
> 10, Rue Vauquelin
> 75231 PARIS Cedex 05
> FRANCE
>



-- 
Kele Xu, PhD Student of Université Pierre et Marie CURIE
Institute Langevin(Vauquelin Campus)
10, Rue Vauquelin
75231 PARIS Cedex 05
FRANCE

**Comment by thirdwing** [comment]
I am the person who should say sorry.

I think I can give you a temporary fix. Can you run `init.context.default()`  first and your code?

I will fix this soon.

**Comment by kelelexu** [comment]
Error: could not find function "init.context.default"

**Comment by thirdwing** [comment]
Which version of R you are using? Can you send me the data?

**Comment by kelelexu** [comment]
_
platform       x86_64-w64-mingw32
arch           x86_64
os             mingw32
system         x86_64, mingw32
status
major          3
minor          2.1
year           2015
month          06
day            18
svn rev        68531
language       R
version.string R version 3.2.1 (2015-06-18)
nickname       World-Famous Astronaut

​
 test.csv
<https://drive.google.com/file/d/0B4PQ-d-kGAcrei1uV2xzNUVxTHc/view?usp=drive_web>
​​
 train.csv
<https://drive.google.com/file/d/0B4PQ-d-kGAcrOENXQmVlcjIwYlk/view?usp=drive_web>
​

2015-11-26 20:40 GMT+01:00 Qiang Kou (KK) <notifications@github.com>:

> Which version of R you are using? Can you send me the data?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dmlc/mxnet/issues/714#issuecomment-159984168>.
>



-- 
Kele Xu, PhD Student of Université Pierre et Marie CURIE
Institute Langevin(Vauquelin Campus)
10, Rue Vauquelin
75231 PARIS Cedex 05
FRANCE

**Comment by gouchangjiang** [comment]
hi, guys, I have across the same problem in Ubuntu. Did you find any solution? Thanks in advance.

**Comment by kelelexu** [comment]
Hi, I have just re-install the packages using:
install.packages("drat", repos="https://cran.rstudio.com")
drat:::addRepo("dmlc")
install.packages("mxnet")
then it work.

**Comment by gouchangjiang** [comment]
I have solved this by reinstall R package! Maybe the error is caused by installing from Rstudio but not the R CMD INSTALL command.

**Comment by gouchangjiang** [comment]
It works just onetime and then the error occurs again!

**Comment by aravindsiv** [comment]
Facing the same error; just like @gouchangjiang commented, it works just once, and then it occurs again.

**Comment by thirdwing** [comment]
Hi, just make sure all of you met the same `'mx.ctx.internal.default.value' not found` error, right?

I will look into this soon.

**Comment by kelelexu** [comment]
Yes, one potential solution for me is to close R stuido and load the
library again, then it will work for a short time.

2015-11-30 18:51 GMT+01:00 Qiang Kou (KK) <notifications@github.com>:

> Hi, just make sure all of you met the same 'mx.ctx.internal.default.value'
> not found error, right?
>
> I will look into this soon.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dmlc/mxnet/issues/714#issuecomment-160704811>.
>



-- 
Kele Xu, PhD Student of Université Pierre et Marie CURIE
Institute Langevin(Vauquelin Campus)
10, Rue Vauquelin
75231 PARIS Cedex 05
FRANCE

**Comment by thirdwing** [comment]
OK. Can you try to add `ctx=mx.cpu()` in `mx.mlp`, just like

```
mx.mlp(data.matrix(x[nonval.ind,]), y[nonval.ind], ctx=mx.cpu(),
                  hidden_node=c(400,100), 
                  out_node=2, out_activation=softmax,
                  num.round=12, array.batch.size=50, 
                  learning.rate=0.01, momentum=0.5, 
                  dropout = 0.5, activation = tanh,
                  eval.data = val.dat, eval.metric=mx.metric.accuracy)
```

**Comment by kelelexu** [comment]
I have tried it, but the problem remains there.

2015-11-30 19:48 GMT+01:00 Qiang Kou (KK) <notifications@github.com>:

> OK. Can you try to add ctx=mx.cpu() in mx.mlp, just like
>
> mx.mlp(data.matrix(x[nonval.ind,]), y[nonval.ind], ctx=mx.cpu(),
>                   hidden_node=c(400,100),
>                   out_node=2, out_activation=softmax,
>                   num.round=12, array.batch.size=50,
>                   learning.rate=0.01, momentum=0.5,
>                   dropout = 0.5, activation = tanh,
>                   eval.data = val.dat, eval.metric=mx.metric.accuracy)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dmlc/mxnet/issues/714#issuecomment-160721493>.
>



-- 
Kele Xu, PhD Student of Université Pierre et Marie CURIE
Institute Langevin(Vauquelin Campus)
10, Rue Vauquelin
75231 PARIS Cedex 05
FRANCE

**Comment by aravindsiv** [comment]
@thirdwing I tried that too, but it gives me the following error:

Error in mx.model.FeedForward.create(out, X = data, y = label, ctx = device,  : 
  formal argument "ctx" matched by multiple actual arguments

**Comment by thirdwing** [comment]
@aravindsiv Can you also post a **minimal** reproducible case?

**Comment by thirdwing** [comment]
@kelelexu When I run the code you provide, I got

```
> x$month <- as.integer(format(x$Original_Quote_Date, %m))
Error: unexpected input in "x$month <- as.integer(format(x$Original_Quote_Date, %m))"
```

Can you remove other things and only keep `mxnet`-related code?

**Comment by tqchen** [comment]
related to #870 . Close due to inactive status, feel free to reopen if there is update
